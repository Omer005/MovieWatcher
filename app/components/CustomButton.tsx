import React, { useEffect } from "react"
import { Text, TextInput, TouchableOpacity, useColorScheme, View, Image } from "react-native"
import { AMMAG_COLOR, BLACK, GRAY, WHITE } from "./colors";
import { COLOR_PRIMARY } from "../utils/colors";


const MyCustomButton = ({ text, imagesource }) => {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <TouchableOpacity
            style={{
                borderColor: "#A9A9A9",
                borderWidth: 1,
                width: "100%",
                borderRadius: 13,
                paddingLeft: 10,
                height:50,
                display: "flex",
                flexDirection: "row",
                justifyContent: "center",
                alignItems: "center"
            }}
        >

            <Image source={imagesource} style={{height:25,width:25}} />

            <Text style={{ color: "#A9A9A9",paddingLeft:"2%" }}>
                {text}
            </Text>


        </TouchableOpacity>
    )
}

export default MyCustomButton