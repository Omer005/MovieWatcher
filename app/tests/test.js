import React from 'react';
import renderer from 'react-test-renderer'; // Import 'renderer' instead of 'react-test-renderer'
import CustomButton from '../components/CustomButton';
import CustomTextInput from '../components/input';
import MyCustomButton from '../components/CustomButton';

    describe('CustomButton', () => {
        it('renders correctly', () => {
            const tree = renderer.create( // Use 'renderer.create' instead of 'render'
                <CustomButton
                    text="Donate Now"
                    onPress={() => { }}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });


    describe('CustomTextInput', () => {
        it('renders correctly', () => {
            const tree = renderer.create( // Use 'renderer.create' instead of 'render'
                <CustomTextInput
                    placeholder="Search"
                    title="Enter Email"
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });


    describe('MyCustomButton', () => {
        it('renders correctly', () => {
            const tree = renderer.create( // Use 'renderer.create' instead of 'render'
                <MyCustomButton
                    text="Donate Now"
                    onPress={() => { }}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });



