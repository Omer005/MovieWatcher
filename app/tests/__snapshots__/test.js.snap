// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#A9A9A9",
      "borderRadius": 13,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <Image
    style={
      {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Text
    style={
      {
        "color": "#A9A9A9",
        "paddingLeft": "2%",
      }
    }
  >
    Donate Now
  </Text>
</View>
`;

exports[`CustomTextInput renders correctly 1`] = `
<View
  style={{}}
>
  <TextInput
    placeholder="Search"
    placeholderTextColor="#A9A9A9"
    style={
      {
        "borderColor": "#A9A9A9",
        "borderRadius": 13,
        "borderWidth": 1,
        "color": "#A9A9A9",
        "fontSize": 15,
        "paddingLeft": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "color": "red",
        "fontWeight": 600,
      }
    }
  />
</View>
`;

exports[`MyCustomButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#A9A9A9",
      "borderRadius": 13,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <Image
    style={
      {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Text
    style={
      {
        "color": "#A9A9A9",
        "paddingLeft": "2%",
      }
    }
  >
    Donate Now
  </Text>
</View>
`;
